dbtcloud_blueprints/download_logs_artifacts.py

```diff
@@ -0,0 +1,126 @@
+from httprequest_blueprints import execute_request, download_file
+import check_run_status
+import argparse
+import os
+import json
+import code
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--api-key', dest='api_key', required=True)
+    parser.add_argument('--account-id', dest='account_id', required=True)
+    parser.add_argument('--run-id', dest='run_id', required=True)
+    args = parser.parse_args()
+    return args
+
+
+def write_json_to_file(json_object, file_name):
+    with open(file_name, 'w') as f:
+        f.write(
+            json.dumps(
+                json_object,
+                ensure_ascii=False,
+                indent=4))
+
+
+def log_step_details(run_details_response, folder_name):
+    if run_details_response['data']['run_steps'] == []:
+        print(
+            f'No logs to download for run {run_details_response["data"]["id"]}')
+    else:
+        for step in run_details_response['data']['run_steps']:
+            step_id = step['id']
+            execute_request.create_folder_if_dne(f'{folder_name}/responses/')
+            execute_request.create_folder_if_dne(f'{folder_name}/logs/')
+            step_file_name = execute_request.combine_folder_and_file_name(
+                f'{folder_name}/responses/', f'step_{step_id}_response.json')
+            debug_log_name = execute_request.combine_folder_and_file_name(
+                f'{folder_name}/logs/', 'dbt.log')
+            output_log_name = execute_request.combine_folder_and_file_name(
+                f'{folder_name}/logs/', 'dbt_console_output.txt')
+            write_json_to_file(
+                step, step_file_name)
+            with open(debug_log_name, 'a') as f:
+                f.write(step['debug_logs'])
+            with open(output_log_name, 'a') as f:
+                f.write(step['logs'])
```

Comment:
Why are we in append mode for these two files and not the other? I'm assuming it's because that these files will be part of a fleet run and will exist across multiple voyages? 

Is there also a reason that you're writing the step['key'] directly and not with json.dumps like in write_json_to_file?

Reply:
@eriicelsken ['logs'] is the equivelant of console output, so it gets a text file. ['debug_logs'] is what dbt would normally generate as a single text file called `dbt.log`. I can show you the files if you're interested.

We're appending because the number of steps are variable and only referenced by their index, not by the work they're doing. As a result, I didn't think that anyone would want to reference data from a specific step. Instead, I thought it was better to aggregate all of the step logs to one location (what they would see if they ran it in Shipyard).

Reply:
Okay, makes sense.